model/portableElement/storage/PortableElementRegistry.php

```diff
@@ -579,5 +595,13 @@ protected function krsortByVersion(array &$array)
         });
     }
 
+    private function createDataObject(array $component)
+    {
+        $componentId = $component['typeIdentifier'].$component['version'];
```

Comment:
I know it is a backport, but are we sure in this case that `typeIdentifier` and `version` indexes exist and are correct?

Reply:
I believe yes, but I think we may avoid this by introducing the way of a proper distinction arrays with a `component` data, as there are no real need to rely on the specific fields of array.  WDYT about something like `md5(json_encode($component));` as a safer option of `componentId` ?

Reply:
Maybe. You can md5 this, but what if there are objects inside, that cannot be json_encoded? If you json_encode + md5 we are reducing the performance just because we do not know what is inside of `$component`. Could we have a check and make sure they exist, or fallback to an approach as you mentioned in last case? 

I do not want to push more about this issue, since it is a backport and was tested. It is up to you.